code/modules/library/book.dm

```diff
@@ -360,11 +360,14 @@
 	var/amount = 1
 
 /obj/item/book/random/Initialize()
-	..()
-	var/list/books = GLOB.library_catalog.get_random_book(amount, FALSE)
+	. = ..()
+	addtimer(CALLBACK(src, PROC_REF(spawn_books)), 0)
```

Comment:
why a timer with delay 0 instead of an INVOKE_ASYNC?

Reply:
Explained on discord